packages/core/src/dialects/abstract/query-generator-typescript.ts

```diff
@@ -1088,7 +1092,7 @@ Only named replacements (:name) are allowed in literal() because we cannot guara
     throw new Error(`_addLimitAndOffset has not been implemented in ${this.dialect.name}.`);
   }
 
-  bulkDeleteQuery(tableName: TableNameOrModel, options: BulkDeleteQueryOptions): string {
+  bulkDeleteQuery(tableName: TableOrModel, options: BulkDeleteQueryOptions): string {
```

Comment:
Just so that we don't forget; can you add a TODO for line 1100 that using limit with a model definition is not supported (yet)?

Reply:
Even better, I just made it use the model definition instead

Reply:
Did you think of a name for your utility library already?

Reply:
Not at all, do you have ideas?

Reply:
Not really, was just curious